apps/system/flash_test/flash_test.c

```diff
@@ -53,44 +53,105 @@
 /****************************************************************************
  * Included Files
  ****************************************************************************/
-
-#include <tinyara/config.h>
-#include <tinyara/progmem.h>
-#include <sys/stat.h>
-
-#include <unistd.h>
 #include <stdio.h>
+#include <string.h>
 #include <stdlib.h>
+#include <errno.h>
+
 #include <tinyara/fs/mtd.h>
 
 /****************************************************************************
- * Pre-processor Definitions
+ * Definitions
  ****************************************************************************/
+#define BUF_SIZE   1024
 
 /****************************************************************************
- * Private data
+ * Private Data
  ****************************************************************************/
 
 /****************************************************************************
- * Public Functions
+ * Private Functions
  ****************************************************************************/
+static int flash_read(unsigned long addr, int size)
+{
+	int ret;
+	int offset;
+	int buf_idx;
+	int read_size;
+	int remaining;
+	FAR uint8_t *buffer;
+	FAR struct mtd_dev_s *dev_mtd = NULL;
+
+	if (addr == 0 || size <= 0) {
+		printf("Invalid arguments, offset 0x%x, read size %d\n", addr, size);
+		return ERROR;
+	}
+
+	dev_mtd = up_flashinitialize();
```

Comment:
`up_flashinitialize` is also only for flat build. Could you add conditions about this?

Reply:
I already added that condition in Kconfig.
`depends on MTD && BUILTIN_APPS && !BUILD_PROTECTED && !BUILD_KERNEL`